server/app/controllers/users.js

```diff
@@ -0,0 +1,169 @@
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import { User } from '../../models';
+
+module.exports = {
+  /**
+   * Create a new user
+   * @param {Object} req - Request object
+   * @param {Object} res - Response object
+   * @returns {Object} - Returns response object
+   */
+  create(req, res) {
+    return User.create(req.body)
+    .then((user) => {
+      bcrypt.compare(req.body.password, user.password, (error) => {
+        if (error) return res.status(400).send({ error });
+        const token = jwt.sign({
+          message: 'signedUp',
+          userId: user.id,
+          roleId: user.roleId
+        }, process.env.SECRET, { expiresIn: '24h' });
+        res.status(201).json({
+          success: true,
+          message: 'You have successfully signed up!',
+          token,
+          userId: user.id,
+          userEmail: user.email
+        });
+      });
+    })
+    .catch(error => res.status(400).send({ error }));
+  },
+
+  /**
+   * Login a user
+   * @param {Object} req - Request object
+   * @param {Object} res - Response object
+   * @returns {Object} - Returns response object
+   */
+  login(req, res) {
+    return User.findOne({
+      where: {
+        username: req.body.username
+      }
+    })
+    .then((user) => {
+      const correct = bcrypt.compareSync(req.body.password, user.password);
+      if (correct) {
+        const token = jwt.sign({
+          message: 'loggedIn',
+          userId: user.id,
+          roleId: user.roleId
+        }, process.env.SECRET, { expiresIn: '24h' });
+        return res.status(200).json({
+          success: true,
+          message: 'You have successfully signed in!',
+          token,
+          userId: user.id,
+          userEmail: user.email
+        });
+      }
+      return res.status(400).send({ message: 'error' });
```

Comment:
`error` seems vague here, you may want to have a more informative error `message` here.